### PR TITLE
Added "!ask" command to respond with link to https://dontasktoask.com/.

### PR DESCRIPTION
**Pull request by yzsshen**
Added an ask module (crate?) that just returns a static response. New function in messaging.rc checks if the message starts with "!ask" then uses handle_ask with the standard way of responding. This should allow replying like "!request" does.

### CODE REVIEW

File: src/ask.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pub async fn respond_ask() -> anyhow::Result<String> {
+    Ok("Don't ask to ask, just ask! \nhttps://dontasktoask.com/".to_string())
+}
+
+#[cfg(test)]
```
Comment (lissahyacinth) [inline src/ask.rs]: This test can be dropped, as it doesn't help test anything meaningful. I think given the very small scope of the PR, it's okay without new tests

File: src/main.rs [hunk 2]
```diff
@@ -140,6 +142,14 @@ async fn handle_roadmap(ctx: &Context, message: &Message) -> anyhow::Result<()>
...
 }
 
+async fn handle_ask(ctx: &Context, message: &Message) -> anyhow::Result<()> {
+    let response = respond_ask().await?;
+
+    reply_chunked(ctx, message.author.mention(), message.channel_id, response).await?;
+
+    Ok(())
+}
+
```
Comment (lissahyacinth) [inline src/main.rs]:
```suggestion
async fn handle_ask(ctx: &Context, message: &Message) -> anyhow::Result<()> {
    reply_chunked(
        ctx,
        message.author.mention(),
        message.channel_id,
        "Don't ask to ask, just ask! \nhttps://dontasktoask.com/".to_string(),
    )
    .await
}
```

There's a few points here; 
1. We don't need to create a function that's only going to return a string. 
2. We can return the Result from `reply_chunked` - we don't need to use `?` to check the result and then return the Ok.

File: src/main.rs [hunk 3]
```diff
@@ -175,6 +185,10 @@ async fn handle_message(ctx: Context, message: Message) {
         if let Err(e) = handle_roadmap(&ctx, &message).await {
             error!("Failed to create Roadmap due to {e}")
         }
+    } else if messaging::is_message_ask(&message) {
+        if let Err(e) = handle_ask(&ctx, &message).await {
+            error!("Failed to link dontask due to {e}")
```
Comment (lissahyacinth) [inline src/main.rs]:
```suggestion
            error!("Failed to link to don't ask to ask due to {e}")
```

**Review by lissahyacinth — CHANGES_REQUESTED** [review]

**Comment by lissahyacinth** [comment]
Thanks so much for this!

There's a few notes to make this a lot cleaner, then I'm happy to merge it in. Basically we can simplify the code to make it fit better - we don't need the new `ask` module either :)

**Comment by yzsshen** [comment]
Removed ask module and simplified as suggested! Thanks for looking over it!

**Review by lissahyacinth — APPROVED** [review]

**Pull request merged**